### ISSUE TITLE
Only check for well-known files in /etc/kubernetes/manifests, not the whole directory

### ISSUE THREAD

**Issue by luxas** [kubernetes/kubeadm#595]
Right now we check if the `/etc/kubernetes/manifests` directory is empty or not, and if it has some kind of content, kubeadm will fail on preflight checks. That is counter-productive in many situations, what kubeadm really needs to do is to check whether the well-known static pod manifests exist already and fail if and only if some of those exist already (`kube-apiserver.yaml`, `kube-controller-manager.yaml`, `kube-scheduler.yaml`, `etcd.yaml`)

Basically, `DirAvailableCheck` [here](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L863) should instead be four different [`FileAvailableCheck`](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L261)s.

This would be a perfect first contribution to kubeadm :+1: :) !

**Comment by mattkelly** [comment]
I'm up for taking a stab at this this weekend. Seems pretty straightforward!

**Comment by xiangpengzhao** [comment]
I just wonder what if there are already manifests (for any reason) named such as `apiserver.yaml` in that dir?

**Comment by stealthybox** [comment]
@xiangpengzhao they're ignored by `kubeadm`.

Both the static-pod and self-hosted strategies explicitly create pods per component from paths derived from `kubeadm.constants`.

**ex.**
static-pods: https://github.com/kubernetes/kubernetes/blob/release-1.9/cmd/kubeadm/app/phases/controlplane/manifests.go#L50-L68
selfhosted: https://github.com/kubernetes/kubernetes/blob/release-1.9/cmd/kubeadm/app/phases/selfhosting/selfhosting.go#L75-L90
